src/benchmarks/micro/ValuesGenerator.cs

```diff
@@ -87,12 +94,40 @@ private static T GenerateValue<T>(Random random)
                 return (T)(object)random.Next();
             if (typeof(T) == typeof(double))
                 return (T)(object)random.NextDouble();
+            if (typeof(T) == typeof(float))
+                return (T)(object)(float)random.NextDouble();
             if (typeof(T) == typeof(string))
                 return (T) (object) GenerateRandomString(random, 1, 50);
             
             throw new NotImplementedException($"{typeof(T).Name} is not implemented");
         }
 
+        private static T GenerateValue<T>(Random random, T minValue, T maxValue)
+        {
+            // This is currently used for many of the System.Math and System.MathF perf tests.
+            //
+            // These functions often have special handling in the JIT to either cache the computed
+            // result or to perform constant folding. This allows us to generate a unique input
+            // per iteration that is within the allowed input domain for a given method.
+
+            if (typeof(T) == typeof(int))
+                return (T)(object)random.Next((int)(object)minValue, (int)(object)maxValue);
+            if (typeof(T) == typeof(double))
+                return (T)(object)BitConverter.Int64BitsToDouble(
```

Comment:
Maybe we could use a simpler approach? From [SO](https://stackoverflow.com/a/1064907/5852046)

`random.NextDouble() * (maximum - minimum) + minimum;`

Reply:
I believe the current approach gives the best distribution of results, over the desired range, without falling into a pit of failure due to IEEE rounding and result representation issues.

Reply:
I am sure it is. However, to understand this code one needs an expert knowledge. You have it, others (including me) don't ;)

Could you add a comment explaining how it works and some unit tests to make sure it's correct?